### PR TITLE
refactor: deep merge config_edn

### LINKED ISSUE #1584: load athens config from env var
All parameters needed by the Athens server should be configurable via a `config.edn` file, whose location is specified through a env var for ease of use with docker.

### PR DESCRIPTION
**Pull request by juniusfree**
closes #1584

### CODE REVIEW

File: src/clj/athens/self_hosted/components/config.clj [hunk 1]
```diff
@@ -13,8 +13,11 @@
 
   (start
     [component]
-    (log/info "Starting configuration component")
-    (assoc component :config (cfg/reload-env)))
+    (let [config        (cfg/reload-env)
+          merged-config (cfg/merge-maps config (:config-edn config))]
+      (log/info "Starting configuration component")
+      (log/info merged-config)
```
Comment (filipesilva) [inline src/clj/athens/self_hosted/components/config.clj:19]: Nit: this isn't needed anymore, the final config is already printed elsewhere.

**Review by filipesilva — APPROVED** [review]
Lgtm, but would like @neotyk opinion on removing the previous convir var. Think this is an overall better approach.

File: README.self-hosted.md [hunk 2]
```diff
@@ -18,7 +18,7 @@ This will create `target/athens-lan-party-standalone.jar`.
 ### Custom Datahike DB path
 
 You can customize where Datahike stores it's DB via
-environment variable `datahike_store_path`.
+environment variable `config_edn`.
```
Comment (neotyk) [inline README.self-hosted.md:21]: That's not just DB path anymore, maybe worth mentioning it.

**Review by neotyk — APPROVED** [review]

File: src/clj/athens/self_hosted/components/config.clj (hunk 1 above)
Comment (juniusfree) [reply src/clj/athens/self_hosted/components/config.clj:19]: There is no other `log/info` for config other than this.

Comment (filipesilva) [reply src/clj/athens/self_hosted/components/config.clj:19]: Thought I had seen repeat prints while we were debugging, lgtm then

**Pull request merged**